src/sentry/options/manager.py

```diff
@@ -81,53 +105,78 @@ def set(self, key, value):
         )
 
         try:
-            self.update_cached_value(key, value)
+            self.update_cached_value(opt.cache_key, value)
+            return True
         except Exception:
             self.logger.warn(CACHE_UPDATE_ERR, key, exc_info=True)
+            return False
+
+    def lookup_key(self, key):
+        try:
+            return self.registry[key]
+        except KeyError:
+            # HACK: Historically, Options were used for random adhoc things.
+            # Fortunately, they all share the same prefix, 'sentry:', so
+            # we special case them here and construct a faux key until we migrate.
+            if key[:7] == 'sentry:':
+                self.logger.info('Using legacy key: %s', key, exc_info=True)
+                # History shows, there was an expectation of no types, and empty string
+                # as the default response value
+                return Key(key, '', object, self.FLAG_DEFAULT, self._make_cache_key(key))
+            raise UnknownOption(key)
 
     def get(self, key):
```

Comment:
I don't have a lot of historical context here, but throwing an exception here seems safer. I'm not sure if that would cause issues with any legacy configuration that is more tolerant of invalid values, though.